sierralocal/jsonwriter.py

```diff
@@ -126,15 +127,45 @@ def format_validation_results(self, validated):
         validation_results = [{'level': v[0], 'message': v[1]} for v in validated]
         return validation_results
 
-    def format_drug_resistance(self, scores, gene):
+    def format_drug_resistance(self, scores, sequence_name, gene, ambiguous, names):
         """
         Returns formatted drug resistance and score breakdowns,
         meant for results output.
         @param scores: dict, results of one query from score_alg.score_drugs()
         @param gene: str, gene found in the sequence
+        @param ambiguous: dict, {sequence name: position of NNN NAs}
+        @param names: list, [sequence names]
         @return drug_resistance: dict, one dictionary encoding
         scores and descriptions
         """
+        # remove all ambiguous NNN positions
+        for drug, info in scores.items():
+            new = [[] for i in info[2]]
+            scores[drug] = list(scores[drug])
+            inds = set() # hold positions for valid scores
+
+            # not fastest way, but works around not mutating while iterating
+            for index, position in enumerate(info[2]):
+                for ind2, mut in enumerate(position):
+                    if int(mut[1:-1]) in ambiguous[sequence_name][gene]:
+                        inds.add(index)
+                    else:
+                        new[index].append(mut)
+
+            new1 = []
+            for i in new:
+                if i:
+                    new1.append(i)
+
+            new2 = []
```

Comment:
same here:
`new2 = [score for index, score in enumerate(info[1]) if index not in indxs]`
also need an inline comment to explain what is going on here